x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/CCRFeatureSet.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ccr;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.xpack.core.XPackFeatureSet;
+import org.elasticsearch.xpack.core.XPackField;
+import org.elasticsearch.xpack.core.XPackSettings;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+public class CCRFeatureSet implements XPackFeatureSet {
+
+    private final boolean enabled;
+    private final XPackLicenseState licenseState;
+    private final ClusterService clusterService;
+
+    @Inject
+    public CCRFeatureSet(Settings settings, @Nullable XPackLicenseState licenseState, ClusterService clusterService) {
+        this.enabled = XPackSettings.CCR_ENABLED_SETTING.get(settings);
+        this.licenseState = licenseState;
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    public String name() {
+        return XPackField.CCR;
+    }
+
+    @Override
+    public String description() {
+        return "Cross Cluster Replication";
+    }
+
+    @Override
+    public boolean available() {
+        return licenseState != null && licenseState.isCcrAllowed();
+    }
+
+    @Override
+    public boolean enabled() {
+        return enabled;
+    }
+
+    @Override
+    public Map<String, Object> nativeCodeInfo() {
+        return null;
+    }
+
+    @Override
+    public void usage(ActionListener<XPackFeatureSet.Usage> listener) {
+        MetaData metaData = clusterService.state().metaData();
+
+        int numberOfFollowerIndices = 0;
+        long lastFollowerIndexCreationDate = 0L;
+        for (IndexMetaData imd : metaData) {
+            if (imd.getCustomData("ccr") != null) {
```

Comment:
nit: maybe use `CCR_CUSTOM_METADATA_KEY` constant instead?

Reply:
Unfortunately this constant is not available from the core module.

Reply:
ok.